Source/GDProject/Player/GDPlayerPawn.cpp

```diff
@@ -106,12 +106,21 @@ void AGDPlayerPawn::RequestUnitAction()
 	if (AGDUnit* Unit = Cast<AGDUnit>(SelectedTile->GetTileElement()))
 	{
 		bWaitingForActionCompletion = true;
+		SelectedUnit = Unit;
```

Comment:
L'unità potresti salvarla in SelectedUnit direttamente all'interno di SelectTile() nel caso contenga un'unità.
Per questo specifico caso non cambia nulla, ma se in futuro ci servirà l'unità selezionata prima di effettuare azioni, lo teniamo già pronto